pkg/controller/common/reconciler.go

```diff
@@ -49,8 +65,25 @@ func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci network
 	return nil
 }
 
-func (r *BaseIngressReconciler) deleteRoutes(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
+func routeMap(routes routev1.RouteList) map[string]*routev1.Route {
+	mp := map[string]*routev1.Route{}
```

Comment:
nit: (Just to get into the habit of doing this)

```suggestion
	mp := make(map[string]*routev1.Route, len(routes.Items))
```

Allocates the map to the needed size immediately.